Remotion/Web/Development.WebTesting.IntegrationTests/DropDownMenuControlObjectTest.cs

```diff
@@ -299,6 +305,77 @@ public void TestHasSupplementalButtonType ()
       Assert.That(dropDownMenu.GetButtonType(), Is.EqualTo(ButtonType.Supplemental));
     }
 
+    [Category("Screenshot")]
+    [Test]
+    public void ScrollIntoView_WithLongDropDownMenu_ScrollInvisibleItemIntoView ()
+    {
+      ScreenshotTestingDelegate<FluentScreenshotElement<DropDownMenuControlObject>> test = (builder, target) =>
+      {
+        builder.AnnotateBox(target.GetMenu(), Pens.Magenta, WebPadding.Inner);
+
+        builder.AnnotateBox(target.SelectItem().WithIndex(95).ScrollIntoView(), Pens.Chartreuse, WebPadding.Inner);
+        builder.AnnotateBox(target.SelectItem().WithIndex(92).ScrollIntoView(), Pens.Red, WebPadding.Inner);
+
+        builder.Crop(target.GetMenu(), new WebPadding(1));
+      };
+
+      var home = Start();
+      try
+      {
+        ResizeBrowserWindowTo(new Size(600, 400));
+        var fluentDropDownMenu = home.DropDownMenus().GetByLocalID("MyDropDownMenu_ManyMenuItems").ForControlObjectScreenshot();
+        fluentDropDownMenu.OpenMenu();
+        Thread.Sleep(250);
+
+        Helper.RunScreenshotTestExact<FluentScreenshotElement<DropDownMenuControlObject>, DropDownMenuControlObjectTest>(
+            fluentDropDownMenu,
+            ScreenshotTestingType.Both,
+            test);
+      }
+      finally
+      {
+        home.Driver.MaximiseWindow(home.Scope);
+      }
+    }
+
+    [Category("Screenshot")]
+    [Test]
+    public void ScrollIntoView_WithLongDropDownMenu_ScrollsBackUp ()
+    {
+      ScreenshotTestingDelegate<FluentScreenshotElement<DropDownMenuControlObject>> test = (builder, target) =>
+      {
+        builder.AnnotateBox(target.GetMenu(), Pens.Magenta, WebPadding.Inner);
+
+        target.SelectItem().WithIndex(90).ScrollIntoView();
+        builder.AnnotateBox(target.SelectItem().WithIndex(2).ScrollIntoView(), Pens.Chartreuse, WebPadding.Inner);
+
+        builder.Crop(target.GetMenu(), new WebPadding(1));
+      };
+
+      var home = Start();
+      try
+      {
+        ResizeBrowserWindowTo(new Size(600, 400));
+        var fluentDropDownMenu = home.DropDownMenus().GetByLocalID("MyDropDownMenu_ManyMenuItems").ForControlObjectScreenshot();
+        fluentDropDownMenu.OpenMenu();
+        Thread.Sleep(250);
```

Comment:
ScrollIntoView already does a `Thread.Sleep`. If we need a second one here, this feels like something that invalidates the API: Do we need this because `OpenMenu` takes a long time to build the dom and update the ui due to the long list of items? If so, this means everyone using this API with a long menu could run into the same issue.

Reply:
This sleep was part of the test `Remotion.Web.Development.WebTesting.IntegrationTests.ScreenshotTest.DropDownMenu` I used as reference, which was added during the original implementation of the fluent screenshotting feature - it should not be necessary anymore with the scrolling API I think, I'll test it. If it still is an issue, some way of waiting for the menu to be opened should rather be added.

Reply:
Great. So, code cleanup if possible. 

And given this delay also needed for non-screenshot tests, we may need a separate ticket to handle the OpenMenu/delay-requirement?

Reply:
I actually didn't find the delay in other tests, just screenshot tests.